### ISSUE TITLE
Next.js should error when it detects a CommonJS transform in Babel config

### ISSUE THREAD

**Issue by Timer** [vercel/next.js#9253]
When using next.js with typescript and importing react `import * as React from 'react';` I'm getting double React imports. Any advice? I'm looking into how some other plugins detect it to see if there is a workaround.

_Originally posted by @cranberyxl in https://github.com/zeit/next.js/pull/8350#issuecomment-547964729_

**Comment by Timer** [comment]
cc @developit

**Comment by cranberyxl** [comment]
I opened a potential workaround in #9254 Let me know if this is a good direction to take. As I note in the PR, I'm trying to convert an existing CRA and there me be some libraries that don't play well together. I will look into a reduced example of it happening.

**Comment by cranberyxl** [comment]
I have isolated the cause to `@babel/plugin-transform-modules-commonjs` which we were previously using. I will see if we can get by without it.

**Comment by cranberyxl** [comment]
Example: https://github.com/cranberyxl/nextjs-double-react

```
$ yarn dev
yarn run v1.19.1
$ next dev
[ wait ]  starting the development server ...
[ info ]  waiting on http://localhost:3000 ...
[ info ]  bundled successfully, waiting for typecheck results...
[ ready ] compiled successfully - ready on http://localhost:3000
[ wait ]  compiling ...
[ info ]  bundled successfully, waiting for typecheck results...
[ ready ] compiled successfully - ready on http://localhost:3000
[ event ] build page: /next/dist/pages/_error
[ wait ]  compiling ...
[ info ]  bundled successfully, waiting for typecheck results...
[ ready ] compiled successfully - ready on http://localhost:3000
[ event ] build page: /
[ wait ]  compiling ...
> Using external babel configuration
> Location: "[REDACTED]/double-react/.babelrc"
[ error ] ./pages/index.tsx 23:7
Module parse failed: Identifier 'React' has already been declared (23:7)
File was processed with these loaders:
 * ./node_modules/next/dist/build/webpack/loaders/next-babel-loader.js
You may need an additional loader to handle the result of these loaders.
| var _jsxFileName = "[REDACTED]/double-react/pages/index.tsx";
| import _JSXStyle from "styled-jsx/style";
> import React from "react";
| var __jsx = React.createElement;
|```

**Comment by developit** [comment]
Hi @cranberyxl - are you able to clarify why you were manually injecting `@babel/plugin-transform-modules-commonjs`? That's done automatically by Next in the cases where it's needed.

**Comment by cranberyxl** [comment]
I imported the babel config from our existing project we are converting from CRA to next.js. I didn't see `@babel/plugin-transform-modules-commonjs` listed in the docs as an existing plugin, but I've found the place in the code now. I don't know if it's possible to parse the custom `.babelrc` and error on existing plugins. Thanks for pointing that out, I'll consider this closed.

**Comment by developit** [comment]
@cranberyxl it's possible, and I think there may be a solution coming soon for doing just that.

/cc @prateekbh regarding babelrc conformance assertion

**Comment by timneutkens** [comment]
This is solved with the new compiler 🚀

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
